app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ListsController < ApplicationController
+  before_action :authorize_request
+  
+  def show_users_lists
+    @user = User.where(id: params[:id]).first
```

Comment:
You already have the `@user` since `authorize_request` returns `@current_user`. You can skip all this logic and just get the lists by doing `@lists = @current_user.lists`. This will ensure that just the lists of the authorized user are returned.